Source/Moq.Analyzers/MoqAsMethodDescriptor.cs

```diff
@@ -0,0 +1,32 @@
+﻿namespace Moq.Analyzers;
+
+/// <summary>
+/// A class that, given a <see cref="SemanticModel"/> and a <see cref="MemberAccessExpressionSyntax"/>, determines if
+/// it is a call to the Moq `Mock.As()` method.
+/// </summary>
+internal class MoqAsMethodDescriptor : MoqMethodDescriptorBase
+{
+    private const string MethodName = "As";
+
+    public override bool IsMatch(SemanticModel semanticModel, MemberAccessExpressionSyntax memberAccessSyntax, CancellationToken cancellationToken)
+    {
+        if (!IsFastMatch(memberAccessSyntax, MethodName.AsSpan()))
+        {
+            return false;
+        }
+
+        ISymbol? symbol = semanticModel.GetSymbolInfo(memberAccessSyntax, cancellationToken).Symbol;
+
+        if (symbol is not IMethodSymbol methodSymbol)
+        {
+            return false;
+        }
+
+        if (!IsContainedInMockType(methodSymbol))
+        {
+            return false;
+        }
+
+        return methodSymbol.Name.AsSpan().SequenceEqual(MethodName.AsSpan()) && methodSymbol.IsGenericMethod;
```

Comment:
It's an (increasingly complicated) "it depends".

In this particular case it shouldn't matter, as `MethodName` is a const `string` and `methodSymbol.Name` appears to already be an allocated string under the hood.

So doing `string.Equals` in theory could be event a bit faster because of the ReferenceEquality check first, but I don't think methodSymbol.Name will ever in practice have reference equality with MethodName.

This particular comparison I like a bit better because it's a bit clearer what's happening, and it matches what's in `MoqMethodDescriptorBase::IsFastMatch()` which does take a `ReadOnlySpan<char>` (the Span-ification begins!), and thus is better to avoid allocs.